python/ray/data/dataset_pipeline.py

```diff
@@ -700,6 +682,325 @@ def __iter__(self):
 
         return EpochDelimitedIterator(self, max_epoch)
 
+    def map(
+        self,
+        fn: RowUDF,
+        *,
+        compute: Union[str, ComputeStrategy] = None,
+        **ray_remote_args,
+    ) -> "DatasetPipeline[U]":
+        """Apply ``Dataset.map`` to each dataset/window in this pipeline."""
+        return self.foreach_window(
+            lambda ds: ds.map(fn, compute=compute, **ray_remote_args)
+        )
+
+    def map_batches(
+        self,
+        fn: BatchUDF,
+        *,
+        batch_size: Optional[int] = 4096,
+        compute: Union[str, ComputeStrategy] = None,
+        batch_format: str = "native",
+        fn_args: Optional[Iterable[Any]] = None,
+        fn_kwargs: Optional[Dict[str, Any]] = None,
+        fn_constructor_args: Optional[Iterable[Any]] = None,
+        fn_constructor_kwargs: Optional[Dict[str, Any]] = None,
+        **ray_remote_args,
+    ) -> "DatasetPipeline[U]":
+        """Apply ``Dataset.map_batches`` to each dataset/window in this pipeline."""
```

Comment:
Any references to the `Dataset` APIs should be single backticks in order for them to serve as API reference links in the rendered docs.
```suggestion
        """Apply `Dataset.map_batches` to each dataset/window in this pipeline."""
```

Reply:
Good catch, I was following what we did before. Wondering is there any way for me to check docs still work before merging this change?

Reply:
Figured out the syntax, updated, it's actually needs `:py:meth:` prefix.

Reply:
Hmm weird, I thought that the single backticks would work since we changed the default role to `:py:obj:` 🤔 

This looks fine though, and for future reference (doesn't need to change in this PR), `:py:` is our default Sphinx domain, so `:meth:` should suffice. And I'll look into why the default role `:py:obj:` isn't working as expected.

Reply:
btw the current doc which has the syntax of single backtick (https://docs.ray.io/en/master/data/package-ref.html#datasetpipeline-api) is not clickable.

I also saw usage of `:py:meths` in other places - https://github.com/ray-project/ray/blob/master/python/ray/data/dataset.py#L146-L149 .